l10n_it_account/models/res_partner.py

```diff
@@ -0,0 +1,10 @@
+import re
+
+from odoo import models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def format_vat_it(self, vat):
+        return vat and re.sub(r"\W+", "", vat).upper() or False
```

Comment:
Da come viene chiamato questo metodo in `base_vat` (https://github.com/odoo/odoo/blob/0268986b60477c7fb46d93f45f3355f546f0155c/addons/base_vat/models/res_partner.py#L545), pare che qui ci debbano essere solo cifre.
Potresti quindi modificare l'espressione regolare in modo da rimuovere tutto ciò che non è una cifra?
Attualmente inserendo la P.IVA `IT1234567s0017`, questa non viene _ripulita_ ma viene sollevato l'errore
![image](https://user-images.githubusercontent.com/32064796/133639986-bb63dff8-cf40-4d8b-bbcf-38b08b64ccae.png)

Questo perché l'espressione regolare in pratica rimuove solo tutto ciò che non è alfanumerico lasciando i caratteri alfabetici.

Essendo fuori dallo scope di questa PR (rimuovere solo gli spazi), questo non lo valuterei come bloccante per il merge ma solo un _nice to have_.

Reply:
Trovo corretta l'eccezione. Rimuovere gli spazi (che la 99% vuol dire lo spazio tra IT e il numero) è una cosa di formattazione del campo: "IT 12345670017" e "IT12345670017" sono la stessa piva formattata in due modi diversi.

Se l'utente inserisce un errore di battitura, trovo giusto segnalarlo e non inserire un euristica per "correggere" l'input per es. rimuovendo le lettere.